utils/gas-utils/gas_utils.go

```diff
@@ -8,12 +8,14 @@ import (
 	"math/big"
 
 	"github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/subnet-evm/precompile/contracts/warp"
 )
 
 const (
-	ReceiveCrossChainMessageStaticGasCost           uint64 = 2_000_000
-	ReceiveCrossChainMessageGasCostPerAggregatedKey uint64 = 1_000
-	ReceiveMessageGasLimitBufferAmount              uint64 = 100_000
+	// The total gas cost for a single call to receiveCrossChainMessage is ~500_000. Add 300_000 to the
+	// static gas cost charged by the Warp precompile per signature verification (200_000) to arrive at this number.
+	ReceiveCrossChainMessageStaticGasCost uint64 = 300_000 + warp.GasCostPerSignatureVerification // Totals 500_000
```

Comment:
Sorry, I'm still a little confused as to why we add `warp.GasCostPerSignatureVerification`, but then add `uint64(numSigners) * warp.GasCostPerSignatureVerification` in `CalculateReceiveMessageGasLimit`.

Reply:
aha! Perfect, thanks :)